R/linkMultipleVariants.R

```diff
@@ -0,0 +1,294 @@
+            }
+            ellfwd[indexVfwd]
+        }
+    }))
+
+    ## Check that the number of Vs in the combined run agrees with the number of
+    ## separate runs
+    if (!combinedDigestParams$mergeForwardReverse) {
+        if (nbrFwdV + nbrRevV != length(lengthsV)) {
+            stop("The number of variable sequences in the combined run does not ",
+                 "correspond to the number of separate runs")
+        }
+    } else {
+        if ((nbrFwdV != nbrRevV) || (nbrFwdV != length(lengthsV))) {
+            stop("The number of variable sequences in the combined run does not ",
+                 "correspond to the number of separate runs")
+        }
+    }
+
+    ## --------------------------------------------------------------------- ##
+    ## Combined quantification - just tabulate combined variable sequences
+    ## --------------------------------------------------------------------- ##
+    outCombined <- do.call(digestFastqs, combinedDigestParams)
+
+    ## Get count matrix with "raw" (uncorrected) sequences
+    countCombined <- outCombined$summaryTable %>%
+        dplyr::select(.data$sequence, .data$nbrReads)
+
+    ## If applicable, separate into forward and reverse sequences
+    ## (separated by _). Also check that all sequences are the same (and
+    ## correct) length
+    if (nbrFwdV > 0 && nbrRevV > 0 && !combinedDigestParams$mergeForwardReverse) {
+        countCombined <- countCombined %>%
+            tidyr::separate(sequence, into = c("forward", "reverse"), sep = "_")
+        if (!all(nchar(countCombined$forward) ==
+                 sum(lengthsV[seq_len(nbrFwdV)]))) {
+            stop("Forward sequences have the wrong length")
+        }
+        if (!all(nchar(countCombined$reverse) ==
+                 sum(lengthsV[nbrFwdV + seq_len(nbrRevV)]))) {
+            stop("Reverse sequences have the wrong length")
+        }
+    } else if (nbrFwdV > 0) {
+        countCombined <- countCombined %>%
+            dplyr::rename(forward = sequence)
+        if (!all(nchar(countCombined$forward) ==
+                 sum(lengthsV[seq_len(nbrFwdV)]))) {
+            stop("Forward sequences have the wrong length")
+        }
+    } else if (nbrRevV > 0) {
+        countCombined <- countCombined %>%
+            dplyr::rename(reverse = sequence)
+        if (!all(nchar(countCombined$reverse) ==
+                 sum(lengthsV[nbrFwdV + seq_len(nbrRevV)]))) {
+            stop("Reverse sequences have the wrong length")
+        }
+    }
+
+    ## Split forward and reverse sequences, respectively, into the parts
+    ## corresponding to the separate runs
+    if ("forward" %in% colnames(countCombined)) {
+        countCombined <- countCombined %>%
+            tidyr::separate(forward, into = paste0("V", seq_len(nbrFwdV)),
+                            sep = cumsum(lengthsV[seq_len(nbrFwdV - 1)]))
+    } else {
+        colnames(countCombined)[colnames(countCombined) == "forward"] <- "V1"
+    }
```

Comment:
Here I am confused: The else-block will be for `! "forward" %in% colnames(countCombined)`, so what will the `"V1"` be assigned to? Is `colnames(countCombined) == "forward"` not going to be all `FALSE`?

The same is true for the `"reverse"` case (lines 253-260

Reply:
Oh, you're right, this doesn't make sense...it's a remnant of a previous logic. The idea was to check whether there is a forward sequence, and then if it needs to be split or not (if not, just rename it V1 for downstream consistency).

Reply:
I think I will leave that for now (you probably know better how to correct it)